server/src/main/java/org/opensearch/action/admin/indices/stats/CommonStatsFlags.java

```diff
@@ -151,6 +167,14 @@ public Flag[] getFlags() {
         return flags.toArray(new Flag[0]);
     }
 
+    public EnumSet<CacheType> getIncludeCaches() {
+        return includeCaches;
+    }
+
+    public String[] getLevels() {
+        return levels;
```

Comment:
These `levels` are specific to cache stats, right? How do the other stat levels work?

Reply:
In theory any stats could want to aggregate by some levels, and could access them here or in ToXContent's params, but after checking it seems like only NodesIndicesStats and this PR actually do. The levels work for aggregation in NodesIndicesStats. 

We added levels to these flags because Ankit realized it was wasteful for each data node to serialize and send all the unaggregated data, before the coordinator node has to aggregate each response itself. He raised an issue for this [here](https://github.com/opensearch-project/OpenSearch/issues/13340) and I think NodesIndicesStats will eventually be changed to also use the levels in these flags to send a pre-aggregated response to the coordinator node.

Reply:
Ahh -- I saw his issue, and now I see how it fits together.

Yeah, applying the leveling logic on each node (before sending back to the coordinator) makes a lot of sense. Thanks!